core/instance.go

```diff
@@ -226,19 +227,32 @@ func (i *instance) isVirtualizationCompatible(spotVirtualizationTypes []string)
 	return false
 }
 
-func (i *instance) isAllowed(instanceType string, allowedList []string) bool {
+func (i *instance) isAllowed(instanceType string, allowedList []string, disallowedList []string) bool {
+	debug.Println("Checking allowed/disallowed list")
+
 	if allowedList != nil && allowedList[0] != "" {
 		for _, a := range allowedList {
 			if a == instanceType {
 				return true
 			}
 		}
+		debug.Println("Instance has been excluded since it was not in the allowed instance types list")
 		return false
+	} else if disallowedList != nil && disallowedList[0] != "" {
```

Comment:
If I'm not mistaken, Split will return you an empty element if none is provided. So entering something like: `<space>t2.medium` would be enough to make that method bug, due to the fact that the first element is checked, but is empty. [Small example](https://play.golang.org/p/JlqyRFLlpQn) in playground.
You might want to use https://golang.org/pkg/strings/#FieldsFunc instead to build list without empty elements.

Reply:
Also another question, as the disallowed instance types in the CLI is general to all ASG and cannot be overridden, what if I had:
* allowedInstancesTypes: [ "t2.medium", "t2.large" ]
* disallowedInstanceTypes: [ "t2.medium" ]

Then t2.medium would be considered a valid choice because you have an `else if`. So the disallowedInstanceTypes would be ignored; this is fine if we defined it as such, but to me it would have made more sense to build up a list of only allowed instances based on the single tag (allowed / disallowed) or both if present.

In my example we'd end up only with a list of allowed "instances" which would be [ "t2.large" ]
And I do like the idea of globbing :+1:

Reply:
Sure, in my opinion the options should be mutually exclusive where `allowed-instsance-types` takes priority. Explicit whitelists should take priority here. I can't think of a situation where someone would want to use both, nor would I encourage that because it's incredibly confusing. Even in your example, why not just remove `t2.medium` from the `allowed-intance-types` option?

In regards to your first comment about value parsing, I agree, but I followed the pattern set by the `allowed-instance-types` option. Since the options are inverse of each other their behavior should be the same. I did not want to deviate, nor is it within the scope of this PR to change the `allowed-instance-types` behavior. If we want to correct that we should correct both in a separate PR.

Reply:
Yes, let's fix them both in another PR.

Reply:
For the white/black listing, that was a suggestion. I'm not sure of use-cases either, I just thought that in a global configuration you could set at a region level certain instance types allowed, and some disallowed at ASG levels.

If you have some ASG where you don't care too much about the type maybe.